apps/rush-lib/src/logic/PackageChangeAnalyzer.ts

```diff
@@ -125,19 +134,59 @@ export class PackageChangeAnalyzer {
     //  });
     // }
 
-    // Determine the current variant from the link JSON.
-    const variant: string | undefined = this._rushConfiguration.currentInstalledVariant;
-
-    // Add the shrinkwrap file to every project's dependencies
-    const shrinkwrapFile: string = path.relative(
-      this._rushConfiguration.rushJsonFolder,
-      this._rushConfiguration.getCommittedShrinkwrapFilename(variant)
-    ).replace(/\\/g, '/');
+    if (
+      this._rushConfiguration.packageManager === 'pnpm' &&
+      this._rushConfiguration.pnpmOptions.incrementalBuildDependencyGranularity === 'project'
+    ) {
+      const projects: RushConfigurationProject[] = [];
+      const projectDependencyManifestPaths: string[] = [];
+
+      for (const project of this._rushConfiguration.projects) {
+        const dependencyManifestFilePath: string = PnpmProjectDependencyManifest.getFilePathForProject(project);
+        const relativeDependencyManifestFilePath: string = path.relative(
+          this._rushConfiguration.rushJsonFolder,
+          dependencyManifestFilePath
+        ).replace(/\\/g, '/');
+
+        if (!FileSystem.exists(dependencyManifestFilePath)) {
+          throw new Error(
```

Comment:
Will **rush build** crash on a clean repo?

Reply:
If `rush link` hasn't been run with this version of rush and with the `"project"` granularity configuration, yes. That's by design, though.